docs/DeveloperGuide.adoc

```diff
@@ -767,15 +825,122 @@ Use case ends.
 +
 Use case resumes at step 2.
 
+[discrete]
+=== Use case: Update person
+
+*MSS*
+
+1.  User find target person by his name.
+2.  AddressBook shows the result of find command.
+3.  User types new information for the target person.
+4.  AddressBook updates the information and displays successful message.
++
+Use case ends.
+
+*Extensions*
+
+[none]
+* 2a. The target person cannot be found.
++
+Use case ends.
+
+* 3a. User types invalid information
++
+[none]
+** 3a1. AddressBook shows an error message.
++
+Use case resumes at step 2.
+
+* 3b. New information is the same as original.
++
+[none]
+** 3b1. AddressBook shows an unnecessary update message.
++
+Use case ends.
+
+[discrete]
+=== Use case: Add a person
+*MSS*
+
+1. User enters values for a new person.
+2. AddressBook add the new person into the database.
++
+Use case ends.
+
+*Extensions*
+2a. There is already this person
++
+Use case ends.
+[none]
+* 2a. The list is empty.
++
+Use case ends.
+
+* 3a. The given index is invalid.
++
+[none]
+** 3a1. AddressBook shows an error message.
++
+Use case resumes at step 2.
+
+[discrete]
+=== Use case: Add a tag to a person
+
+*MSS*
+
+1. User enters values for a new person
+2. AddressBook add the new person into the database
++
+Use case ends.
+
+*Extensions*
+
+[none]
+* 2a. There is already this person
++
+Use case ends.
+
+[discrete]
+=== Use case: Show help window
+
+*MSS*
+
+1. User enters "help" for the help command
+2. AddressBook displays the help window
++
+Use case ends.
+
+*Extensions*
+
+* 1a. User presses F1 on the keyboard
++
+Use case resumes at step 2.
+
+* 1b. User clicks on help icon
++
+User case resumes at step 2.
```

Comment:
You should show unexpected scenarios and how the issue is handled.

Perhaps you can condense all the methods of how the user access the help function into step 1.